sql-plugin/src/main/scala/com/nvidia/spark/rapids/higherOrderFunctions.scala

```diff
@@ -487,10 +494,23 @@ case class GpuTransformKeys(
       }
       withResource(newKeysCol) { newKeysCol =>
         withResource(GpuMapUtils.replaceExplodedKeyAsView(arg.getBase, newKeysCol.getBase)) {
-          updatedMapView =>
+          updatedMapView => {
             GpuMapUtils.assertNoNullKeys(updatedMapView)
-            GpuMapUtils.assertNoDuplicateKeys(updatedMapView)
-            GpuColumnVector.from(updatedMapView.copyToColumnVector(), dataType)
+            withResource(updatedMapView.dropListDuplicatesWithKeysValues()) { deduped =>
+              if (exceptionOnDupKeys) {
+                // Compare child data row count before and after removing duplicates to determine
+                // if there were duplicates.
+                withResource(deduped.getChildColumnView(0)) { a =>
+                  withResource(updatedMapView.getChildColumnView(0)) { b =>
+                    if (a.getRowCount != b.getRowCount) {
+                      throw GpuMapUtils.duplicateMapKeyFoundError
+                    }
+                  }
+                }
+              }
+              GpuColumnVector.from(updatedMapView.copyToColumnVector(), dataType)
```

Comment:
Is there a reason we are using the original input and not the deduped input here? We use deduped when creating a map. This code should still work because I think the read picks the last matching key, but I am not 100% sure on that.

Reply:
I think this was an oversight. I will change this and see what other tests can be added to make show that we really are using LAST_WIN.